### DISCUSSION TITLE
Prompts disappears on closing tab

### DISCUSSION THREAD

**Discussion by akashdiphazra** [ghostty-org/ghostty#4018] [Issue Triage]
https://github.com/user-attachments/assets/9923ef31-e7ff-4233-8ce3-1e135f5e1f51

OS : Fedora 41 Gnome
Prompt : Bash and Starship
# Ghostty version : 
![Screenshot From 2024-12-30 12-34-50](https://github.com/user-attachments/assets/07255e1f-fdcb-4d59-a835-b39c52e5fb51)
It happening version : 1.0.0 also

**Comment by 00-kat** [comment]
Try disable shell integration and see if that changes anything.
```ini
shell-integration = none
```

**Reply by karnaKali** [reply]
that was bugging me. but fixed my problem.

**Reply by 00-kat** [reply]
Technically it's not a “fix”, but rather just a workaround.

**Comment by akashdiphazra** [comment]
Ok, This is working but , if don't add this option it will behave same

**Reply by 00-kat** [reply]
In that case it's likely to be a bug with the prompt redrawing that the shell integration does.

**Comment by neniuk** [comment]
This also happens to me, and setting `shell-integration = none` in the config indeed seems to be a partial workaround. It also seems to happen on resizing the terminal — however, it seems that even with `shell-integration = none` there is still some strange behavior going on with the shell prompt. Attached is an example video, comparing **Ghostty** (left) to **Gnome Terminal** (right).

[example.webm](https://github.com/user-attachments/assets/c0457b56-5174-4bd3-9033-15a86c4ebf9e)

**Reply by neniuk** [reply]
Also if not setting `shell-integration = none`, on resize the upper part of the shell prompt will completely disappear, and will not reappear until you run another command or restart the terminal.

**Comment by ghuh** [comment]
I'm running into a similar issue with starship where just the first line of a two line prompts disappears.  It seems to happen if I take focus away from Ghostty too long (e.g. 10+ minutes) or open/close splits.

I originally [opened a ticket](https://github.com/starship/starship/issues/7140) on starship, but I'm going to close it because I think this is a Ghostty bug.

This is in Ghostty v1.2.3

**Reply by ghuh** [reply]
```
shell-integration = none
```
Seems to "fix" the issue for me as well

**Comment by ghuh** [comment]
This is still an issue in the most recent starship release, v1.24.2.  I'm still pretty sure this is a Ghostty bug.

**Comment by eduardolat** [comment]
this is also happening to me, i'm on gnome, the `shell-integration=none` don't work for me

**Comment by ghuh** [comment]
I think this may have been fixed as of Ghostty v1.3.1.  I at least can reproduce it by opening and closing splits anymore.
